core/src/message_queue.c

```diff
@@ -82,23 +93,27 @@ int MESSAGE_QUEUE_push(MESSAGE_QUEUE_HANDLE handle, MESSAGE_HANDLE element)
     int result;
     if (handle == NULL || element == NULL)
     {
+        /*Codes_SRS_MESSAGE_QUEUE_17_007: [ MESSAGE_QUEUE_push shall return a non-zero value if handle or element are NULL. ]*/
         LogError("invalid argument - handle(%p), element(%p).", handle, element);
         result = __LINE__;
     }
     else
     {
-
 		MESSAGE_QUEUE_STORAGE* temp = (MESSAGE_QUEUE_STORAGE*)malloc(sizeof(MESSAGE_QUEUE_STORAGE));
         if (temp == NULL)
         {
+            /*Codes_SRS_MESSAGE_QUEUE_17_009: [ MESSAGE_QUEUE_push shall return a non-zero value if any system call fails. ]*/
             LogError("malloc failed.");
             result = __LINE__;
         }
         else
         {
             DList_InitializeListHead((PDLIST_ENTRY)temp);
             temp->message = element;
+            /*Codes_SRS_MESSAGE_QUEUE_17_010: [ A successful call to MESSAGE_QUEUE_push on an empty queue will cause the message queue to be non-empty (MESSAGE_QUEUE_is_empty shall return false). ]*/
+            /*Codes_SRS_MESSAGE_QUEUE_17_011: [ Messages shall be pushed into the queue in a first-in-first-out order. ]*/
             DList_AppendTailList((PDLIST_ENTRY)&(handle->queue_head), (PDLIST_ENTRY)temp);
+            /*Codes_SRS_MESSAGE_QUEUE_17_008: [ MESSAGE_QUEUE_push shall return zero on success. ]*/
```

Comment:
Yes and no. It returns void so I don't have a return value to check. Obviously, it's manipulating pointers, so something could go wrong, but this unit is the owner of the queue, so it shouldn't go wrong.